File: easy/two-sum/cpp/two_sum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

vector<int> twoSum(const vector<int>& nums, int target) {
    unordered_map<int,int> m;
    for (int i = 0; i < (int)nums.size(); ++i) {
        int want = target - nums[i];
        auto it = m.find(want);
        if (it != m.end()) return {it->second, i};
        m[nums[i]] = i;
    }
    return {};
}

File: easy/two-sum/typescript/twoSum.ts
export function twoSum(nums: number[], target: number): number[] {
  const seen = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const want = target - nums[i];
    if (seen.has(want)) return [seen.get(want) as number, i];
    seen.set(nums[i], i);
  }
  return [];
}
